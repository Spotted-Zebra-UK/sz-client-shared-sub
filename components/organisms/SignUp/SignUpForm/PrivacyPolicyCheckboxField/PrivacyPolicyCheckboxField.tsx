import React, { FC } from 'react';
import CheckboxFormField from '../../../../molecules/CheckboxFormField/CheckboxFormField';

interface PrivacyPolicyCheckboxField {
  value: boolean;
  onChange: (value: boolean, name: string) => void;
  name: string;
  id: string;
  error?: string;
}

const PrivacyPolicyCheckboxField: FC<PrivacyPolicyCheckboxField> = props => {
  const renderPrivacyPolicyCheckboxLabel = () => {
    return (
      <span className="SignUpForm__PrivacyPolicyCheckboxLabel">
        I have read the{' '}
        <a
          className="SignUpForm__PrivacyPolicyCheckboxLabel__Link"
          href="https://www.spottedzebra.co.uk/privacy/privacy"
          target="_blank"
          rel="noopener noreferrer"
        >
          Privacy Notice
        </a>
        , and I am an adult.
      </span>
    );
  };

  return (
    <CheckboxFormField label={renderPrivacyPolicyCheckboxLabel()} {...props} />
  );
};

export default PrivacyPolicyCheckboxField;
