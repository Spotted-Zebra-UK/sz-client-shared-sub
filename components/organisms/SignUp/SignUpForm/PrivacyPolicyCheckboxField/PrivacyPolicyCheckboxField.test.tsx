import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import PrivacyPolicyCheckboxField from './PrivacyPolicyCheckboxField';

describe('PrivacyPolicyCheckboxField', () => {
  it('should render correctly', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <PrivacyPolicyCheckboxField
        value={true}
        onChange={onChange}
        name="test-name"
        id="test-id"
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
