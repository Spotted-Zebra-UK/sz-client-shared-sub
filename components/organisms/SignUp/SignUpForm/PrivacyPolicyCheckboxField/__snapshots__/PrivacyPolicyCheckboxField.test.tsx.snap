// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicyCheckboxField should render correctly 1`] = `
<CheckboxFormField
  id="test-id"
  label={
    <span
      className="SignUpForm__PrivacyPolicyCheckboxLabel"
    >
      I have read the
       
      <a
        className="SignUpForm__PrivacyPolicyCheckboxLabel__Link"
        href="https://www.spottedzebra.co.uk/privacy/privacy"
        rel="noopener noreferrer"
        target="_blank"
      >
        Privacy Notice
      </a>
      , and I am an adult.
    </span>
  }
  name="test-name"
  onChange={[MockFunction]}
  value={true}
/>
`;
